src/main/java/com/aviva/CarRecommendations/InterestFilter.java

```diff
@@ -0,0 +1,65 @@
+package com.aviva.CarRecommendations;
+
+import com.aviva.DataAccess.BankingDataProcess;
+import com.aviva.Entities.AccountHolder;
+import com.aviva.Entities.Car;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class InterestFilter {
+
+    public JSONObject getBestFiveCars(String accountNumber) {
+        BankingDataProcess bdpInit = new BankingDataProcess();
+        AccountHolder user = bdpInit.getAccountHolder(accountNumber);
+
+        EstimateBudget ebInit = new EstimateBudget();
+        float yearlyBudget = ebInit.calculateYearlyBudget(user);
+        user.setSavings(yearlyBudget);
+        float monthlyBudget = ebInit.calculateMonthlyBudget(user);
+        user.setMonthlyBudget(monthlyBudget);
+
+        BudgetFilter bfInit = new BudgetFilter();
+        ArrayList<Car> cars = bfInit.getRecommendedCars(user);
+
+        SensoRate srInit = new SensoRate();
+        ArrayList<Car> bestFive = new ArrayList<>();
+        float[] rates = new float[10];
+
+        for (int i = 0; i < cars.size(); i++) {
+            float interestRate = srInit.getInterestRate(user, cars.get(i));
+            rates[i] = interestRate;
+        }
+
+        float[] sortedRates = Arrays.copyOf(rates, 10);
+        Arrays.sort(sortedRates);
+
+        for (int i = 0; i < 5; i++) {
+            float sortedRate = sortedRates[i];
+            int carIndex = getFirstIndex(rates, sortedRate);
```

Comment:
I feel like this could have been done in a slightly better way. Have you considered using a HashMap?

Reply:
I agree with Leen's on this because HashMap might be more efficient.

Reply:
efficiency can be dealt with later when we have more time